x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/MlMemoryTracker.java

```diff
@@ -250,15 +250,11 @@ public Long getJobMemoryRequirement(String taskName, String id) {
             return null;
         }
 
-        if (MlTasks.TRAINED_MODEL_DEPLOYMENT_TASK_NAME.equals(taskName)) {
-            return getTrainedModelTaskMemoryRequirement(id);
-        } else {
-            Map<String, Long> memoryRequirementByJob = memoryRequirementByTaskName.get(taskName);
-            if (memoryRequirementByJob == null) {
-                return null;
-            }
-            return memoryRequirementByJob.get(id);
+        Map<String, Long> memoryRequirementByJob = memoryRequirementByTaskName.get(taskName);
```

Comment:
I think this won't work properly as we've never `put` the memory requirement for an allocation into `memoryRequirementByTaskName`. It is nice not to have to cache the memory for model allocations as we can access their memory requirement from the cluster state directly and thus, don't have to worry about clearing the entry from the map here.

Reply:
> I think this won't work properly as we've never put the memory requirement for an allocation into memoryRequirementByTaskName.

We never call this method asking for model size estimations.

Reply:
I think this also fits in with "model allocations are not jobs".

This method is for getting the memory requirement for a _job_, so it would be wrong to call it for a trained model.

The first line of the method could assert that `taskName` is one of the task names we use for things we call "jobs".  Then hopefully a test will fail if someone tries to use this too generically in the future.